deepnote_core/resources/scripts/resource_usage.py

```
@@ -0,0 +1,256 @@
+    def _send_text(self, code: int, text: str) -> None:
+        self._send_response(code, text.encode(), "text/plain")
+
+    def _send_response(self, code: int, body: bytes, content_type: str) -> None:
+        self.send_response(code)
+        self.send_header("Content-Type", content_type)
+        self.send_header("Content-Length", str(len(body)))
+        self.send_header("X-Content-Type-Options", "nosniff")
+        self.end_headers()
+        self.wfile.write(body)
+
+    def log_message(self, format: str, *args: Any) -> None:
```

Ruff (0.14.11)

Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)